snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml

```diff
@@ -8,6 +8,50 @@ device:
 
 sysobjectid: 1.3.6.1.4.1.318.1.*
 
+metadata:
+  device:
+    fields:
+      vendor:
+        value: "apc"
+      serial_number:
+        symbol:
+          MIB: POWERNET-MIB
+          OID: 1.3.6.1.4.1.318.1.1.1.1.2.3.0
+          name: upsAdvIdentSerialNumber
+          # The serial number of the UPS internal microprocessor.  This number is set at
+          # the factory. NOTE: This number does NOT correspond to the serial number on the rear of the UPS.
```

Comment:
Shouldn't they match?

Reply:
> NOTE: This number does NOT correspond to the serial number on the rear of the UPS.
This comment is from the MIB, and TBH, it's a bit confusing. Maybe the SN on rear of the UPS is another kind of SN not necessarily tied to the device itself (but not completely sure about this).

About `upsAdvIdentSerialNumber`, after some research, it matches the `SN: XXX` that is present in sysDescr and I think it should be the SN user expect to see.

Reply:
There another location where the serial number seems to be present when looking at SNMP Walks (BAsensored being the serial number in this case):

```
.1.3.6.1.2.1.1.1.0 = STRING: "APC Web/SNMP Management Card (MB:v3.9.2 PF:v3.9.2 PN:apc_hw02_aos_392.bin AF1:v3.7.2 AN1:apc_hw02_sumx_372.bin MN:AP9619 HR:A10 SN: BAsensored MD:12/04/2007) (Embedded PowerNet SNMP Agent SW v2.2 compatible)"
...
.1.3.6.1.4.1.318.1.4.2.4.1.1.1 = INTEGER: 1
.1.3.6.1.4.1.318.1.4.2.4.1.1.2 = INTEGER: 2
.1.3.6.1.4.1.318.1.4.2.4.1.2.1 = STRING: "BAsensored"
.1.3.6.1.4.1.318.1.4.2.4.1.2.2 = STRING: "BAsensored"
.1.3.6.1.4.1.318.1.4.2.4.1.3.1 = STRING: "apc_hw02_aos_392.bin"
.1.3.6.1.4.1.318.1.4.2.4.1.3.2 = STRING: "apc_hw02_sumx_372.bin"
.1.3.6.1.4.1.318.1.4.2.4.1.4.1 = STRING: "v3.9.2"
.1.3.6.1.4.1.318.1.4.2.4.1.4.2 = STRING: "v3.7.2"
```

But this seems not reliable (https://oidref.com/1.3.6.1.4.1.318.1.4 is called experimental) and the structure is a table, making it harder collect. It seems there are different SN for the OS (aos) and the application module (SUMX), but I think we are more interested in the OS (AOS) serial number, that should be the one present in the sysDescr.